lib/src/config/route/app_routes.dart

```diff
@@ -0,0 +1,71 @@
+import 'package:flutter/cupertino.dart';
+
+import '../../core/util/constants.dart' as constants;
+import '../../domain/repository/i_movie_repository.dart';
+import '../../domain/usecase/implementation/fetch_movies_by_genre_usecase.dart';
+import '../../domain/usecase/implementation/fetch_movies_from_endpoint_usecase.dart';
+import '../../domain/usecase/implementation/fetch_valid_genres_usecase.dart';
+import '../../presentation/view/movies_by_genres_view.dart';
+import '../../presentation/view/movies_view.dart';
+
+class AppRoutes {
+  final IMovieRepository repository;
+
+  AppRoutes(this.repository);
+
+  Map<String, Widget Function(BuildContext)>? endpointRoutes;
+
+  Map<String, Widget Function(BuildContext)> get routes {
+    final map = <String, Widget Function(BuildContext)>{};
+    map.addAll(endpointRoutes!);
+    map.addAll(genreRoute);
+    return map;
+  }
+
+  Map<String, Widget Function(BuildContext)> get genreRoute => {
+        constants.genreAppRoute: (context) => MoviesByGenreView(
+              fetchMoviesByGenreUsecase: FetchMoviesByGenreUsecase(
+                repository: repository,
+              ),
+              fetchValidGenresUseCase: FetchValidGenresUseCase(
+                repository: repository,
+              ),
+            ),
+      };
+
+  Map<String, Widget Function(BuildContext)> getMovieEndpointRoutes() =>
+      endpointRoutes ??= {
+        constants.endpointPopular: (context) => MoviesView(
+              fetchMoviesUseCase: FetchPopularUseCase(
+                repository: repository,
+              ),
+              fetchValidGenresUseCase: FetchValidGenresUseCase(
+                repository: repository,
+              ),
+            ),
+        constants.endpointNowPlaying: (context) => MoviesView(
+              fetchMoviesUseCase: FetchNowPlayingUseCase(
+                repository: repository,
+              ),
+              fetchValidGenresUseCase: FetchValidGenresUseCase(
+                repository: repository,
+              ),
+            ),
+        constants.endpointUpcoming: (context) => MoviesView(
+              fetchMoviesUseCase: FetchUpcomingUseCase(
+                repository: repository,
+              ),
+              fetchValidGenresUseCase: FetchValidGenresUseCase(
+                repository: repository,
+              ),
+            ),
+        constants.endpointTopRated: (context) => MoviesView(
+              fetchMoviesUseCase: FetchTopRatedUseCase(
+                repository: repository,
+              ),
+              fetchValidGenresUseCase: FetchValidGenresUseCase(
```

Comment:
You can remove the fetch from the names, because that is an action and you don't need to specify that, you can replace with moviesUseCase, genresUseCase